src/zcash/Note.cpp

```diff
@@ -41,20 +42,31 @@ uint256 SproutNote::nullifier(const SproutSpendingKey& a_sk) const {
 }
 
 // Construct and populate Sapling note for a given payment address and value.
-SaplingNote::SaplingNote(const SaplingPaymentAddress& address, const uint64_t value) : BaseNote(value) {
+SaplingNote::SaplingNote(
+    const SaplingPaymentAddress& address,
+    const uint64_t value,
+    unsigned char _leadByte
+) : BaseNote(value) {
     d = address.d;
     pk_d = address.pk_d;
-    librustzcash_sapling_generate_r(r.begin());
+    leadByte = _leadByte;
```

Comment:
It may be worth asserting here that it is either `0x01` or `0x02`.

Reply:
right now, we're checking leadbyte validity in `TransactionBuilder` + whenever we attempt to decrypt the plaintext. we could add a sanity check when instantiating the Note, but it won't be height-dependent.

Reply:
Yeah, it's just an added assertion so it would only be as a sanity check for later consumers of the API.